pkg/tlscheck/tlscheck.go

```diff
@@ -18,9 +19,21 @@ const (
 	timeout = 2 * time.Second
 )
 
+// addrValid validates the URL. It assumes scheme prefixes have been removed
+func addrValid(addr string) error {
+	addrNoScheme := urlfmt.TrimHTTPPrefixes(addr)
+	// url.Parse requires scheme to trigger the correct variant of parsing (it has two)
+	_, err := url.Parse("https://" + addrNoScheme)
+	return err
+}
```

Comment:
nit: `urlfmt.TrimHTTPPrefixes` is redundant here, already done in CheckTLS.

```suggestion
func addrValid(addrNoScheme string) error {
	// url.Parse requires scheme to trigger the correct variant of parsing (it has two)
	_, err := url.Parse("https://" + addrNoScheme)
	return err
}
```

or perhaps do the trim and return both the sanitized address and error here:

```go
func addrValid(addr string) (string, error) {
        addrNoScheme := urlfmt.TrimHTTPPrefixes(addr)

	// url.Parse requires scheme to trigger the correct variant of parsing (it has two)
	_, err := url.Parse("https://" + addrNoScheme)
	return addrNoScheme, err
}

func CheckTLS(ctx context.Context, origAddr string) (bool, error) {
        addr, err := addrValid(origAddr)
	if err != nil {
		return false, err
	}
        ...
}
```

Reply:
Yes, that is correct. I was more thinking about making that function self-contained and testable. I can update the code and the test cases to reflect your proposal.

Reply:
I assumed that was why, whatever you decide OK by me.

Reply:
The problem is that `addrValid("http://exam ple.com:80/abc")` returns no error - i.e., `url.Parse("https://http://exam ple.com:80/abc")` returns no error... I think I prefer the double stripping of scheme 😅

Reply:
Perhaps this then 😉  (50% joking)

```go
for strings.HasPrefix(addr, "http://") || strings.HasPrefix("https://") {
   addr := urlfmt.TrimHTTPPrefixes(addr)
}
```

Reply:
Actually, we are only checking here `http://` and `https://`, but there are other prefixes as well. I will add a check for that.

Reply:
I changed the validation to be a bit stricter in [f472ed4](https://github.com/stackrox/stackrox/pull/11668/commits/f472ed43bd7e89a45cdb62fb8c78bf1aa1559d6d)

Reply:
(I will resolve that and move on, but feel free to add comments - we can always do some follow-ups to this PR)